### ISSUE TITLE
Preconditions.where with STATE field is missing autocompletion for state values

### ISSUE THREAD

**Issue by anna-geller** [kestra-io/kestra#6104]
### Feature description

![image](https://github.com/user-attachments/assets/f68b60c7-83d9-452c-806d-6cca82f1c444)

![image](https://github.com/user-attachments/assets/8f37e2fe-5c3f-4bb5-b615-236ca74e03bf)

should be the same as:

![image](https://github.com/user-attachments/assets/d985890b-6e65-4b38-8fb2-a79c373866d1)

**Comment by loicmathieu** [comment]
Yes, it's because value/values are dynamic and non-typed inside the where filter.
There is no link between the field and the type of the value attribute in a where filter.

**Comment by anna-geller** [comment]
Brian, we need you as our JSON schema and autocompletion expert 😄

**Comment by brian-mulier-p** [comment]
It can be tackled with generic classes instead of enum @loicmathieu, it's a bit of extra classes though...

**Comment by brian-mulier-p** [comment]
But if we want it to be dynamic it's harder to handle. It can take time...

**Comment by loicmathieu** [comment]
Preconditions has been removed in 2.0 in favor of `dependsOn` and a new `when` condition
